### DISCUSSION TITLE
When loading a glb file, the colors of the materials are changed upon loading

### DISCUSSION THREAD

**Discussion by Miyasaka-prono82** [atteneder/glTFast#677] [Q&A]
Hi,
The color of the cube created is (R,G,B) = (65,110,50) as shown in Figure 1, but the glb model loaded with GLTFast becomes (R,G,B) = (138,175.122) as shown in Figure 2. 
Figure 1
![Figure 1](https://github.com/atteneder/glTFast/assets/56023011/23c54683-0ec5-4be2-9f8b-937c2abf4aaa)

Figure 2 (Import GLTFast)
![Figure 2](https://github.com/atteneder/glTFast/assets/56023011/67a65283-cb1b-4c26-9313-5740978fe5de)

When dragged and dropped into Unity, it already appears with the color of Figure 2.
 I would like to load it with the same color as created (Figure 1). 
Are there any settings required for this? Please let me know if additional information is needed.

**Comment by Miyasaka-prono82** [comment]
This is the glb file I was using.
[Cube.zip](https://github.com/atteneder/glTFast/files/14293886/Cube.zip)

**Reply by SadiAtBim** [reply]
gltFast importer apply its own shader to materials of glbs. You need to check the glb material and shader and gltF shader material. It should be same. 
I think you need to recheck your glb creator side when export. 

![image](https://github.com/atteneder/glTFast/assets/129041312/24569166-34be-4fb7-80f8-e665b803a4c3)

**Reply by Miyasaka-prono82** [reply]
I exported a Cube created in a project with GLTFast installed. 
When I loaded the exported Cube, its color had already changed. 
Is there a way to make it the same color as when it was created?

The cube on the left was created in Unity. The cube on the right is the left cube exported and imported.
![ExportCube](https://github.com/atteneder/glTFast/assets/56023011/d1cbed2f-fa06-4bec-a0f2-1b28f3158af0)

**Reply by SadiAtBim** [reply]
Ok do you have shader variant in graphics settings as per doc of GLTFast? Secondly, how to exporting it? i mean which one from runtime export or editor export?

**Reply by Miyasaka-prono82** [reply]
The shader variant is set as shown in the diagram.
![shader variant](https://github.com/atteneder/glTFast/assets/56023011/87e9816a-9f04-4451-83e7-4f09f1c955ef)

What I want to do this time is import. The problem of colors changing occurs even with glb files created outside of GLTFast. I believe this is not an issue with GLTFast's export, but rather an issue with GLTFast's import.

**Reply by SadiAtBim** [reply]
Import and export are two different things. gltfast import glb in a way that it creates object and material on its own shader. While export is not STABLE as i know 
https://github.com/atteneder/glTFast/issues/351

Might be possible they will work on it. but i will recommend you (for now) Use UnityGLTF export

**Reply by Miyasaka-prono82** [reply]
I am troubled by the issue that the color of the imported model differs from the color of the model when created. Is there any solution?

**Comment by atteneder** [comment]
Hi,

There's been an issue #680 that the base color was exported in an incorrect color space that was fixed in [glTFast 6.7.0](https://docs.unity3d.com/Packages/com.unity.cloud.gltfast@6.7/changelog/CHANGELOG.html#670---2024-06-25). Please go ahead and update your package.

Thanks
